[ZEN-4339] Implement optional 2nd phase of validation with KZOP

ghillairet opened the pull request:
This PR adds an advanced validation mode for OpenAPI 3 documents. The advanced validation is provided by KaiZen Parser (KZOP). The advanced validation is only performed when the advanced option is passed to the OpenAPi3Validator constructor (which it is now true by default).

This PR should not be merged until we update the target platform to include latest version of KZOP and JsonOverlay. Those are present in this PR but are added as jars in the openapi3 project lib folder.

ghillairet committed:
[ZEN-4339] Implement optional 2nd phase of validation with KZOP

ghillairet committed:
[ZEN-4339] Implement optional 2nd phase of validation with KZOP

ghillairet committed:
[ZEN-4339] Implement optional 2nd phase of validation with KZOP

Add Validation preference page for OpenAPI v3. The page contains the option to use the advanced validation from KZOP. This option is enable by default.

ghillairet committed:
Fix formatting

tfesenko commented on com.reprezen.swagedit.core/META-INF/MANIFEST.MF:
```diff
@@ -9,7 +9,7 @@ Require-Bundle: org.eclipse.ui,
  org.eclipse.jface.text,
  org.dadacoalition.yedit,
- com.google.guava;bundle-version="15.0.0",
+ com.google.guava;bundle-version="21.0.0",
```
@ghillairet , we use Guava 16 in the API Studio, see https://github.com/ModelSolv/RepreZen/commit/42f0f2eb0e954dd1c0516e945050ff54c579e901#r30206252 . How was version 21 of Guava chosen?

And... Do we still need Guava, even after migration to Java 8? :)

tfesenko committed:
Revert "[ZEN-4339] Implement optional 2nd phase of validation with KZOP"

This reverts commit c758cef7d21ef1994936e38d820d3e4fbfaea184.

tfesenko committed:
ZEN-4339 - use KaiZen-OpenAPI-Parser from an p2 site

tfesenko committed:
ZEN-4339 - Restore the check for non-empty errors
See
https://github.com/RepreZen/KaiZen-OpenAPI-Editor/pull/457/commits/c758cef7d21ef1994936e38d820d3e4fbfaea184#diff-e4631c2896d91adc9f5ae802883b9bc9

tfesenko commented on com.reprezen.swagedit.core/src/com/reprezen/swagedit/core/json/references/JsonReferenceValidator.java:
```diff
@@ -120,53 +114,23 @@ public void setFactory(JsonSchemaFactory factory) {
-        return errors;
-    }
-
-    protected JsonSchema getSchema(JsonDocument doc, String type, Map<String, JsonSchema> schemas) {
```
This code was extracted to `JsonSchemaValidator `, is that right?

tfesenko commented on com.reprezen.swagedit.openapi3/src/com/reprezen/swagedit/openapi3/validation/OpenApi3Validator.java:
```diff
@@ -10,35 +10,112 @@
+            try {
+                OpenApi3 result = new OpenApi3Parser().parse(baseURI.toURL(), true);
+
+                for (ValidationItem item : result.getValidationResults().getItems()) {
```
Nice!

tfesenko committed:
ZEN-4339 - add missing OSGi dependencies

ghillairet committed:
[ZEN-4339] Implement optional 2nd phase of validation with KZOP

Remove propertyListener from preference store should be done before preference store is deleted.

ghillairet committed:
[ZEN-4339] Implement optional 2nd phase of validation with KZOP

Activate advanced validation when there are no previous errors (not warnings).

ghillairet committed:
[ZEN-4339] Implement optional 2nd phase of validation with KZOP

Update target platform with latest KZOP.

ghillairet committed:
[ZEN-4339] Implement optional 2nd phase of validation with KZOP

Revert guava version to 15.

tfesenko commented on com.reprezen.swagedit.openapi3/src/com/reprezen/swagedit/openapi3/editor/OpenApi3Editor.java:
```diff
@@ -67,8 +67,9 @@ protected YEditSourceViewerConfiguration createSourceViewerConfiguration() {
-        super.dispose();
+        // preference store is removed in AbstractTextEditor.dispose()
         getPreferenceStore().removePropertyChangeListener(advancedValidationListener);
+        super.dispose();
```
Thanks!
